### DISCUSSION TITLE
how to import aiomysql errors

### DISCUSSION THREAD

**Discussion by xrokz** [aio-libs/aiomysql#717] [Q&A] [answered]
i want to make a error handle
if i got an error about wrong db name tell me on shell that i have a db error
but i don't know how to get error codes

**Comment by Nothing4You** [comment, marked as answer]
Hi,

unfortunately this question is quite old and therefore probably not relevant anymore.
aiomysql generally raises the same error types as PyMySQL (where applicable).
for convenience we also provide the PyMySQL error types in the aiomysql package.
in case of an incorrect table name `foo` this raises `pymysql.err.ProgrammingError: (1146, "Table 'mysql.foo' doesn't exist")`.
the error code will be available in `e.args[0]`.

example:
```py3
from pymysql.constants import ER

try:
    await cursor.execute("select 1 from foo")
except aiomysql.ProgrammingError as e:
    if e.args[0] == ER.NO_SUCH_TABLE: # 1146
        print("table doesn't exist")
    else:
        raise e
```
